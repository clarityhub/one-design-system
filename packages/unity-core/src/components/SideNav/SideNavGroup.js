import styled from '@emotion/styled';

const SideNavGroup = styled.div`
    transition: all 0.3s ease-in;

    a,
    a:visited {
        padding-left: 1rem;
    }
`;

export default SideNavGroup;
