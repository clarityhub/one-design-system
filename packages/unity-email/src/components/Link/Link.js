import React from 'react';
import { node, string } from 'prop-types';

const Link = ({ children, href }) => (
	<a href={href}>{children}</a>
);

Link.propTypes = {
	children: node,
	href: string.isRequired,
};

export default Link;
