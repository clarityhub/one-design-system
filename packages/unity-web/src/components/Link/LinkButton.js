import React from 'react';
import { node, shape, func, string } from 'prop-types';

import Button from '../Buttons/Button';

const LinkButton = ({ history, to, onClick, ...rest }) => (
	<Button
		{...rest}
		onClick={(event) => {
			onClick && onClick(event);
			history.push(to);
		}}
	/>
);

LinkButton.propTypes = {
	children: node.isRequired,
	history: shape({
		push: func.isRequired,
	}).isRequired,
	to: string.isRequired,
};

export default LinkButton;