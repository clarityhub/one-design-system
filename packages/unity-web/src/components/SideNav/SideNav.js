import styled from '@emotion/styled';
import { func, node, oneOfType } from 'prop-types';

import { SIDENAV_WIDTH } from './config';

const SideNav = styled.div`
    box-sizing: border-box;
    padding: 3rem 2rem 3rem 0;
    width: ${SIDENAV_WIDTH}rem;
`;

/**
 * Hack for docz and react-docgen
 */
const SideNavProps = () => <div />;
SideNavProps.propTypes = {
	children: oneOfType([node, func]),
};

SideNav.propTypes = SideNavProps.propTypes;
export { SideNavProps };

export default SideNav;
