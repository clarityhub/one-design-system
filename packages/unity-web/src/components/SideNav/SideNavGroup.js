import styled from '@emotion/styled';
import { node } from 'prop-types';

const SideNavGroup = styled.div`
    transition: all 0.3s ease-in;

    a,
    a:visited {
        padding-left: 1rem !important;
    }
`;

/**
 * Hack for docz and react-docgen
 */
const SideNavGroupProps = () => <div />;
SideNavGroupProps.propTypes = {
	children: node,
};

SideNavGroup.propTypes = SideNavGroupProps.propTypes;
export { SideNavGroupProps };

export default SideNavGroup;
