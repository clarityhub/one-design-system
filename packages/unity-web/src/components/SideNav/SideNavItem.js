import styled from '@emotion/styled';
import { css } from '@emotion/core';
import colors from '@clarityhub/unity-core/lib/colors';

import SideNavGroup from './SideNavGroup';

const SideNavItem = styled.div`
    width: 100%;

    ${SideNavGroup} {
        opacity: 0;
        max-height: 0;
        overflow: hidden;
    
        ${({ selected }) => selected && css`
            opacity: 1;
            max-height: 400px;
        `}
    }

    > a,
    > a:visited {
        border: 0;
        text-decoration: none;
        color: ${colors.darkGray.default};
        display: block;
        font-size: 0.9rem;
        padding: 0.6rem 0;
        margin-bottom: 0;

        ${({ selected }) => selected && css`
            color: ${colors.dark.default};
            font-weight: bold;
        `}

        &:hover {
            border: 0;
            text-decoration: none;
            color: ${colors.dark.default};
            display: block;
            margin-bottom: 0;
        }
    }
`;

export default SideNavItem;
