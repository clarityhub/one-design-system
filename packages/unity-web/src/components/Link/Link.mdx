---
menu: Components
name: Link
---

import { Playground, PropsTable } from 'docz';
import Link, { LinkPropTypes } from './Link';

# Link

## \<Link /\> component

```js
import Link from '@clarityhub/unity-web/lib/components/Link';
```

## Prop Types

<PropsTable of={LinkPropTypes} />

## Examples

<Playground>
    <Link href="#">Link</Link>
    <Link type="blockLink" href="#">Block Link</Link>
    <Link href="#" component='div'>Custom Component</Link>
</Playground>