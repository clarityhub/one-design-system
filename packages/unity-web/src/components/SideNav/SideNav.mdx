---
name: SideNav
menu: Components
---

import { Component } from 'react';
import { Playground, PropsTable } from 'docz';
import SideNav from './SideNav';
import SideNavItem from './SideNavItem';
import SideNavGroup from './SideNavGroup';
import Link from '../Link';

# SideNav

The `SideNav` (aka Sidebar) of the application

## \<SideNav /\> component

```js
import SideNav, { SideNavItem, SideNavGroup } from '@clarityhub/unity-core/components/SideNav';
```

## Prop Types

<PropsTable of={SideNav} />

## Examples

<Playground>
    {() => {
        class ControlledSideNav extends Component {
            constructor(props) {
                super(props);

                this.state = {
                    selected: 0,
                }

                this.onSelect = this.onSelect.bind(this);
            }

            onSelect(i) {
                const doChange = function (e) {
                    e.preventDefault();

                    this.setState({
                        selected: i,
                    });
                };

                return doChange.bind(this);
            }

            render() {
                const { selected } = this.state;
                return (
                    <SideNav>
                        <SideNavItem selected={selected === 0}>
                            <Link href="#" onClick={this.onSelect(0)}>Home</Link>
                        </SideNavItem>
                        <SideNavItem selected={selected === 1}>
                            <Link href="#" onClick={this.onSelect(1)}>Insights</Link>
                        </SideNavItem>
                        <SideNavItem selected={selected === 2}>
                            <Link href="#" onClick={this.onSelect(2)}>Integrations</Link>
                        </SideNavItem>
                        <SideNavItem selected={selected === 3}>
                            <Link href="#" onClick={this.onSelect(3)}>Apps</Link>
                        </SideNavItem>
                        <SideNavItem selected={selected === 4}>
                            <Link href="#" onClick={this.onSelect(4)}>Account</Link>

                            <SideNavGroup>
                                <SideNavItem>
                                    <Link href="#">Profile</Link>
                                </SideNavItem>
                                <SideNavItem>
                                    <Link href="#">Billing</Link>
                                </SideNavItem>
                                <SideNavItem>
                                    <Link href="#">Usage</Link>
                                </SideNavItem>
                                <SideNavItem>
                                    <Link href="#">Users</Link>
                                </SideNavItem>
                            </SideNavGroup>
                        </SideNavItem>
                        <SideNavItem selected={selected === 5}>
                            <Link href="#" onClick={this.onSelect(5)}>Developers</Link>

                            <SideNavGroup>
                                <SideNavItem>
                                    <Link href="#">API Keys</Link>
                                </SideNavItem>
                                <SideNavItem>
                                    <Link href="#">Web Console</Link>
                                </SideNavItem>
                                <SideNavItem>
                                    <Link href="#">Webhooks</Link>
                                </SideNavItem>
                                <SideNavItem>
                                    <Link href="#">Logs</Link>
                                </SideNavItem>
                            </SideNavGroup>
                        </SideNavItem>
                    </SideNav>
                );
            }
        }

        return <ControlledSideNav />
    }}
</Playground>