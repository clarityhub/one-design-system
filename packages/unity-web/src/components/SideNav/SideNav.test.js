import React from 'react';
import renderer from 'react-test-renderer';

import Link from '../Link';
import SideNav from './SideNav';
import SideNavItem from './SideNavItem';
import SideNavGroup from './SideNavGroup';

describe('<SideNav />', () => {
	it('renders a sidenav', () => {
		const node = (
			<SideNav>
				<SideNavItem selected>
					<Link href="#">Home</Link>
				</SideNavItem>
				<SideNavItem>
					<Link href="#">Insights</Link>
				</SideNavItem>
				<SideNavItem>
					<Link href="#">Integrations</Link>
				</SideNavItem>
				<SideNavItem>
					<Link href="#">Apps</Link>
				</SideNavItem>
				<SideNavItem>
					<Link href="#">Account</Link>

					<SideNavGroup>
						<SideNavItem>
							<Link href="#">Profile</Link>
						</SideNavItem>
						<SideNavItem>
							<Link href="#">Billing</Link>
						</SideNavItem>
						<SideNavItem>
							<Link href="#">Usage</Link>
						</SideNavItem>
						<SideNavItem>
							<Link href="#">Users</Link>
						</SideNavItem>
					</SideNavGroup>
				</SideNavItem>
				<SideNavItem>
					<Link href="#">Developers</Link>

					<SideNavGroup>
						<SideNavItem>
							<Link href="#">API Keys</Link>
						</SideNavItem>
						<SideNavItem>
							<Link href="#">Web Console</Link>
						</SideNavItem>
						<SideNavItem>
							<Link href="#">Webhooks</Link>
						</SideNavItem>
						<SideNavItem>
							<Link href="#">Logs</Link>
						</SideNavItem>
					</SideNavGroup>
				</SideNavItem>
			</SideNav>
		);

		const tree = renderer.create(node).toJSON();

		expect(tree).toMatchSnapshot();
	});
});