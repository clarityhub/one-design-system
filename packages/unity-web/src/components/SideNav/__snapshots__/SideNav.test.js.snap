// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideNav /> renders a sidenav 1`] = `
.emotion-46 {
  box-sizing: border-box;
  padding: 3rem 2rem 3rem 0;
  width: 14rem;
}

.emotion-1 {
  width: 100%;
}

.emotion-1 .emotion-26 {
  opacity: 0;
  max-height: 0;
  overflow: hidden;
  opacity: 1;
  max-height: 400px;
}

.emotion-1 > a,
.emotion-1 > a:visited {
  border: 0 !important;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #999;
  display: block;
  font-size: 0.9rem;
  padding: 0.6rem 0;
  margin-bottom: 0;
  color: #566ad2;
  font-weight: bold;
}

.emotion-1 > a:hover,
.emotion-1 > a:visited:hover {
  border: 0 !important;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #322439;
  display: block;
  margin-bottom: 0;
}

.emotion-1 > a:hover,
.emotion-1 > a:visited:hover {
  color: #566ad2;
}

.emotion-0 {
  border-bottom: 1px solid #566ad2;
  color: #566ad2;
  display: inline-block;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: 'Open Sans',Arial,sans-serif;
  font-weight: 400;
  string: font-family:'Open Sans',Arial,sans-serif;
  font-weight: 400;
  color: #566ad2;
}

.emotion-0:hover,
.emotion-0:active {
  border-bottom: 2px solid #566ad2;
  border-bottom-width: 2px;
  color: #566ad2;
  margin-bottom: -1px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-0:visited {
  border-bottom: 1px solid rgb(50,72,187);
  color: rgb(50,72,187);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-0:visited:hover,
.emotion-0:visited:active {
  margin-bottom: 0;
}

.emotion-4 {
  width: 100%;
}

.emotion-4 .emotion-26 {
  opacity: 0;
  max-height: 0;
  overflow: hidden;
}

.emotion-4 > a,
.emotion-4 > a:visited {
  border: 0 !important;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #999;
  display: block;
  font-size: 0.9rem;
  padding: 0.6rem 0;
  margin-bottom: 0;
}

.emotion-4 > a:hover,
.emotion-4 > a:visited:hover {
  border: 0 !important;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #322439;
  display: block;
  margin-bottom: 0;
}

.emotion-25 {
  -webkit-transition: all 0.3s ease-in;
  transition: all 0.3s ease-in;
}

.emotion-25 a,
.emotion-25 a:visited {
  padding-left: 1rem !important;
}

<div
  className="emotion-46 emotion-47"
>
  <div
    className="emotion-1 emotion-2"
  >
    <a
      className="emotion-0"
      href="#"
    >
      Home
    </a>
  </div>
  <div
    className="emotion-4 emotion-2"
  >
    <a
      className="emotion-0"
      href="#"
    >
      Insights
    </a>
  </div>
  <div
    className="emotion-4 emotion-2"
  >
    <a
      className="emotion-0"
      href="#"
    >
      Integrations
    </a>
  </div>
  <div
    className="emotion-4 emotion-2"
  >
    <a
      className="emotion-0"
      href="#"
    >
      Apps
    </a>
  </div>
  <div
    className="emotion-4 emotion-2"
  >
    <a
      className="emotion-0"
      href="#"
    >
      Account
    </a>
    <div
      className="emotion-25 emotion-26"
    >
      <div
        className="emotion-4 emotion-2"
      >
        <a
          className="emotion-0"
          href="#"
        >
          Profile
        </a>
      </div>
      <div
        className="emotion-4 emotion-2"
      >
        <a
          className="emotion-0"
          href="#"
        >
          Billing
        </a>
      </div>
      <div
        className="emotion-4 emotion-2"
      >
        <a
          className="emotion-0"
          href="#"
        >
          Usage
        </a>
      </div>
      <div
        className="emotion-4 emotion-2"
      >
        <a
          className="emotion-0"
          href="#"
        >
          Users
        </a>
      </div>
    </div>
  </div>
  <div
    className="emotion-4 emotion-2"
  >
    <a
      className="emotion-0"
      href="#"
    >
      Developers
    </a>
    <div
      className="emotion-25 emotion-26"
    >
      <div
        className="emotion-4 emotion-2"
      >
        <a
          className="emotion-0"
          href="#"
        >
          API Keys
        </a>
      </div>
      <div
        className="emotion-4 emotion-2"
      >
        <a
          className="emotion-0"
          href="#"
        >
          Web Console
        </a>
      </div>
      <div
        className="emotion-4 emotion-2"
      >
        <a
          className="emotion-0"
          href="#"
        >
          Webhooks
        </a>
      </div>
      <div
        className="emotion-4 emotion-2"
      >
        <a
          className="emotion-0"
          href="#"
        >
          Logs
        </a>
      </div>
    </div>
  </div>
</div>
`;
